src/sonic-bgpcfgd/tests/test_sonic-cfggen.py

```diff
@@ -13,7 +13,7 @@ def run_test(name, template_path, json_path, match_path):
     template_path = os.path.join(TEMPLATE_PATH, template_path)
     json_path = os.path.join(DATA_PATH, json_path)
     cfggen = os.path.abspath("../sonic-config-engine/sonic-cfggen")
-    command = ['/usr/bin/python2.7', cfggen, "-T", TEMPLATE_PATH, "-t", template_path, "-y", json_path]
+    command = ['/usr/bin/python3', cfggen, "-T", TEMPLATE_PATH, "-t", template_path, "-y", json_path]
```

Comment:
One suggestion is to completely remove  '/usr/bin/python3' from the command array.
Just cfggen would be enough

Reply:
I considered that, but until we remove all Python 2, I want to explicitly call the Python 3 version, so that it doesn't accidentally try to run the Python 2 version.

Reply:
ok. sounds good.

Reply:
@pavel-shirshov: When attempting to call a sonic-cfggen locally via a relative path, there were missing dependencies. Thus, I made further changes to add a test dependency on the sonic-config-engine package in order to ensure the package is installed in the build environment, and then let the location be resolved via the system paths.

However, there are now some test failures. Can you please investigate them?

Reply:
@tahmed-dev: I'm curious if these test failures may be related to the new sorting differences since you removed the dependency of sonic-cfggen on natsort. Maybe you can comment on this?

Reply:
@jleveque Yes, the errors are related to the removal of natsort from the cfggen. I thought I addressed those errors by make py2/py3 folder until Python 2 is completely removed.

Reply:
@tahmed-dev: Thanks for confirming! Do you have time to look into a solution for this? This PR was opened to unblock yet another PR. There is a chain of dependencies that are now unfortunately blocked on this.

Reply:
@jleveque welcome. is this package still being build for python2 as well?

Reply:
@tahmed-dev: For the time being, yes, we still need both Python 2 and 3 versions of sonic-config-engine.

Reply:
I meant `bgpcfgd` packages as it looks as if the test is repeated/similar to those in `sonic-cfggen` package. @pavel-shirshov are those duplicate or just similar to `sonic-cfggen` tests? The do not have the py2/py3 mods and so they didn't cause issues when I converted the 'sonic-cfggen'. Now since they are using `sonic-cfggen` Python3 similar issues are introduced or so I think.

Reply:
@pavel-shirshov: Can you please answer @tahmed-dev's question above?

Reply:
As I understand, now sonic-cfggen is being run by python3. So dictionary order are different now. That was the reason I run sonic-cfggen in python2.
So I'd suggest first convert sonic-cfggen to python3. By convert I mean regenerate all the file with python3, check that python3 didn't change anything (not that easy to test), and then this PR will go smoothly.

Reply:
btw, @pavel-shirshov , running sonic-cfggen in python2 is causing bgpcfgd failing see issue #5897

Reply:
@lguohan. https://github.com/Azure/sonic-buildimage/pull/5746 wasn't failed for 6 days and now it started?

Reply:
how close are we on this? does joe needs to regenerate all the test results?

Reply:
What is missing currently: dependencies of sonic-config-engine for python3. If you check why mellanox build is failing - because no netaddr installed for python3.

Also I think it is better to move sonic-config-engine to python3 first, Than continue working with this pr.

Reply:
```
File "/sonic/src/sonic-config-engine/sonic-cfggen", line 33, in <module>\n    import netaddr\nImportError: No module named netaddr\n
```

Reply:
That's strange. The netaddr package should get installed along with sonic-config-engine: https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/setup.py#L11

If the Python 2 package of sonic-config-engine is installed, the Python 2 package of netaddr should also be installed; same for Python 3.

Reply:
what is the work amount to do that?

Reply:
@tahmed-dev was looking into it.

Reply:
> I considered that, but until we remove all Python 2, I want to explicitly call the Python 3 version, so that it doesn't accidentally try to run the Python 2 version.
> 
> I think we can clean it up later, once we have removed Python 2 completely.

@jleveque So to confirm, this package is being built only for Python 2.

Reply:
No. @pavel-shirshov has [migrated](https://github.com/Azure/sonic-buildimage/pull/5746) the package to Python 3.

Reply:
@jleveque regenerate the verification data for Python 3 cfggen. I could not push to your branch and so create this [PR:5913](https://github.com/Azure/sonic-buildimage/pull/5913)

Reply:
@tahmed-dev: I have incorporated your changes into my branch. A number of the tests are now passing, but some are still failing. It appears as though you may have used an out-of-date branch. I will try to fix.

Reply:
All check builds are passing now. Thanks for your help, @tahmed-dev!

Reply:
great news! Thanks @jleveque!